### PR TITLE
feat(vue): Rework Vue documentation to include Vue2 and Vue3 support

### PR DESCRIPTION
**Pull request by kamilogorek**
To be released once `6.10.0` is out.

### CODE REVIEW

**Review by HazAT — DISMISSED** [review]
Can we also update the alert on this page to say that we now support Vue 3 but it should be considered Early Access.
image

**Comment by kamilogorek** [comment]
Oh, I meant to remove it and forgot to update commit. Fixed.

File: src/includes/getting-started-config/javascript.vue.mdx [hunk 1]
```diff
@@ -1,26 +1,109 @@
...
 import { Integrations } from "@sentry/tracing";
 
 Sentry.init({
-  Vue: Vue,
+  Vue,
   dsn: "___PUBLIC_DSN___",
-  integrations: [new Integrations.BrowserTracing()],
+  integrations: [
+    new Integrations.BrowserTracing({
+      routingInstrumentation: Sentry.vueRouterInstrumentation(router),
```
Comment (AbhiPrasad) [inline src/includes/getting-started-config/javascript.vue.mdx:15]: Need to define `router` above like we do for Vue 3

File: src/includes/performance/configure-sample-rate/javascript.vue.mdx [hunk 2]
```diff
@@ -41,15 +41,19 @@ Sentry.init({
 });
 ```
 
-### Tracking Child Components
+## Configure Tracing Insturmentation
```
Comment (AbhiPrasad) [inline src/includes/performance/configure-sample-rate/javascript.vue.mdx:44]:
```suggestion
## Configure Tracing Instrumentation
```

File: src/includes/performance/configure-sample-rate/javascript.vue.mdx [hunk 3]
```diff
@@ -41,15 +41,19 @@ Sentry.init({
...
 
-If you want to track child components, and see more details about the rendering process, configure the integration to track them all:
+If you want to track child components or use additional hooks to see more details about the rendering process, you can configure the SDK using options below:
```
Comment (AbhiPrasad) [inline src/includes/performance/configure-sample-rate/javascript.vue.mdx:46]: Can we add a note that this adds spans to the current active transaction on the scope?

**Review by AbhiPrasad — APPROVED** [review]

**Comment by kamilogorek** [comment]
@AbhiPrasad all done

File: src/includes/getting-started-config/javascript.vue.mdx [hunk 4]
```diff
@@ -1,26 +1,115 @@
...
+Sentry.attachErrorHandler(miscApp, { logErrors: true });
 ```
 
-Sentry's Vue SDK accepts a few different configuration options that let you change its behavior:
+The SDK accepts a few different configuration options that let you change its behavior:
 
-- Passing in `Vue` is optional. If you do not pass it, `window.Vue` must be present.
-- Passing in `attachProps` is optional and is `true` if it is not provided. If set to `false`, Sentry will suppress sending all Vue components' props for logging.
-- Passing in `logErrors` is optional and is `false` if it is not provided. If set to `true`, Sentry will call original Vue's `logError` function as well.
+- `attachProps` (defaults to `true`) - include all Vue components' props with the events
```
Comment (imatwawana) [inline src/includes/getting-started-config/javascript.vue.mdx:108]:
```suggestion
- `attachProps` (defaults to `true`) - Includes all Vue components' props with the events.
```

File: src/includes/getting-started-config/javascript.vue.mdx [hunk 5]
```diff
@@ -1,26 +1,115 @@
...
+- `logErrors` (defaults to `false`) - decides whether SDK should call Vue's original `logError` function as well
```
Comment (imatwawana) [inline src/includes/getting-started-config/javascript.vue.mdx:109]:
```suggestion
- `logErrors` (defaults to `false`) - Decides whether SDK should call Vue's original `logError` function as well.
```

File: src/includes/getting-started-config/javascript.vue.mdx [hunk 6]
```diff
@@ -1,26 +1,115 @@
...
+- `trackComponents` (defaults to `false`) - decides whether to track components by hooking into its lifecycle methods. Can be either set to `boolean` to enable/disable tracking for all of them. Or to an array of specific component names (case-sensitive).
```
Comment (imatwawana) [inline src/includes/getting-started-config/javascript.vue.mdx:110]:
```suggestion
- `trackComponents` (defaults to `false`) - Decides whether to track components by hooking into its lifecycle methods. Can be set to either `boolean`, to enable/disable tracking for all of them, or to an array of specific component names (case-sensitive).
```

File: src/includes/getting-started-config/javascript.vue.mdx [hunk 7]
```diff
@@ -1,26 +1,115 @@
...
+- `timeout` (defaults to `2000`) - Time in miliseconds dictating how long to wait until the tracked root activity is marked as finished and sent of to Sentry.
```
Comment (imatwawana) [inline src/includes/getting-started-config/javascript.vue.mdx:111]:
```suggestion
- `timeout` (defaults to `2000`) - Time in milliseconds dictating how long to wait until the tracked root activity is marked as finished and sent off to Sentry.
```

File: src/includes/performance/configure-sample-rate/javascript.vue.mdx [hunk 8]
```diff
@@ -41,15 +41,19 @@ Sentry.init({
 });
 ```
 
-### Tracking Child Components
+## Configure Tracing Instrumentation
 
-If you want to track child components, and see more details about the rendering process, configure the integration to track them all:
+If you want to track child components or use additional hooks to see more details about the rendering process, you can configure the SDK using options below. Captured data will be then stored as spans and attached to the currently active transaction on the scope:
+
+- `trackComponents` (defaults to `false`) - decides whether to track components by hooking into its lifecycle methods. Can be either set to `boolean` to enable/disable tracking for all of them. Or to an array of specific component names (case-sensitive).
```
Comment (imatwawana) [inline src/includes/performance/configure-sample-rate/javascript.vue.mdx:48]:
```suggestion
- `trackComponents` (defaults to `false`) - Decides whether to track components by hooking into its lifecycle methods. Can be set to either `boolean`, to enable/disable tracking for all of them, or to an array of specific component names (case-sensitive).
```

File: src/includes/performance/configure-sample-rate/javascript.vue.mdx [hunk 9]
```diff
@@ -41,15 +41,19 @@ Sentry.init({
...
+- `timeout` (defaults to `2000`) - Time in miliseconds dictating how long to wait until the tracked root activity is marked as finished and sent of to Sentry.
```
Comment (imatwawana) [inline src/includes/performance/configure-sample-rate/javascript.vue.mdx:49]:
```suggestion
- `timeout` (defaults to `2000`) - Time in milliseconds dictating how long to wait until the tracked root activity is marked as finished and sent off to Sentry.
```

File: src/platforms/javascript/guides/vue/configuration/integrations/vue-router.mdx [hunk 10]
```diff
@@ -1,20 +1,22 @@
...
 description: "Learn how Sentry's Vue Tracing Integration allows you to track rendering performance during an initial application load."
+redirect_from:
+  - /sdks/react/integrations/vue-router/
 ---
 
 The Vue Tracing Integration allows you to track rendering performance during an initial application load.
 
-Sentry injects a handler inside Vue's `beforeCreate` mixin, providing access to a Vue component during its life cycle stages.
-When Sentry encounters a component named `root`, which is a top-level Vue instance (as in `new Vue({})`), we use our AM BrowserTracing integration,
+Sentry uses mixins functionality to providing an access to a Vue component during its life cycle stages.
```
Comment (imatwawana) [inline src/platforms/javascript/guides/vue/configuration/integrations/vue-router.mdx:10]:
```suggestion
Sentry uses mixins functionality to provide access to a Vue component during its life cycle stages.
```

File: src/platforms/javascript/guides/vue/configuration/integrations/vue-router.mdx [hunk 11]
```diff
@@ -1,20 +1,22 @@
...
 To do that, you need to specify which components to track and what hooks to listen to (you can find a list of all available hooks [here](https://vuejs.org/v2/api/#Options-Lifecycle-Hooks)). You can also turn on tracking for all the components. However, it may be rather noisy if your app consists of hundreds of components. We encourage being more specific. If you don't provide hooks, Sentry will track a component's `mount` and `update` hooks.
 
 Note that when specifying hooks we use the simple verb rather than `before` and `-ed` pairs. For example, `destroy` is correct. `beforeDestroy` and `destroyed` are incorrect.
 
-To set up the Vue Tracing Integration, you will first need to configure the AM BrowserTracing integration itself. For details on how to do this, see [Performance](/platforms/javascript/guides/vue/performance/).
+To set up the Vue Tracing Integration, you will first need to configure the BrowserTracing integration itself. For details on how to do this, see [Performance](/platforms/javascript/guides/vue/performance/).
```
Comment (imatwawana) [inline src/platforms/javascript/guides/vue/configuration/integrations/vue-router.mdx:19]:
```suggestion
To set up the Vue Tracing Integration, you will first need to configure the BrowserTracing integration itself. For details on how to do this, check out our [Performance documentation](/platforms/javascript/guides/vue/performance/).
```

**Review by imatwawana — APPROVED** [review]
Minor wording changes for formatting consistency and flow. Otherwise LGTM!

**Pull request merged**